classes/zombie.rb

```diff
@@ -1,39 +1,111 @@
 class Zombie
-  attr_accessor :name, :username, :password, :image, :location, :bio, :created_at, :tweets, :next_tweet_id, :stalkers, :prey
+  attr_accessor :name, :username, :password, :image,
+                        :location, :bio, :created_at, :tweets, :stalkers,
+                        :prey, :retweet # :logged_in
 
+  # Class level array of Zombies, used for storing & searching in a few places
+  @@instances = []
+
+  # Run at startup
   def initialize
-    self.next_tweet_id = 0
-    self.tweets = []
-    self.stalkers = []
-    self.prey = []
+#    self.logged_in = false
+    @tweets = []
+    @stalkers = []
+    @prey = []
+    @image = "http://vignette3.wikia.nocookie.net/lego/images/8/81/Zombie_Groom.png/revision/latest?cb=20120823164249"
+    @@instances << self
+  end
+
+  # Getter for the Class attribute instances
+  def self.instances()
+    @@instances
+  end
+
+  # Create class method that accepts username argument.
+  # Returns instance of class with correct username.
+  def self.find_zombie(username)
+    matching_zombie = nil
+    @@instances.each do | zombie |
+      if zombie.username == username
+        matching_zombie = zombie
+      end
+    end
+    matching_zombie
   end
 
+  # Create a new instance of the Tweet class with some content,
+  # and store it in the array of this user's tweets
   def create_tweet(content:, location: nil)
     t = Tweet.new
     t.zombie = self
     t.content = content
     t.location = location
-    t.tweet_id = self.next_tweet_id
-    self.next_tweet_id += 1
-    self.tweets.push(t)
+    @tweets.push(t)
+    t
+  end
+
+  # Create and store a new tweet using an existing tweet's content
+  def retweet(tweet_id)
+    tweet = Tweet.find_tweet(tweet_id)
+    t = create_tweet(content: tweet.content)
+    t.retweeted = true
+    t.original_tweet = tweet
+    tweet.retweets += 1
     t
   end
 
+  # Search through this user's tweets by tweet id and return the matching tweet.
+  # Return nil if no match is found.
+  def return_tweet(tweet_id)
+    return_tweet = nil
+    @tweets.each { | tweet | return_tweet = tweet if tweet.unique_id == tweet_id }
+    return_tweet
```

Comment:
It makes sense to me to call this `find_tweet` and to perhaps see if there's a way to share code between this and the `Tweet` class somehow